pymeasure/instruments/apex/apex1000.py

```diff
@@ -0,0 +1,176 @@
+    SLOTS = range(1,9)
+    DEV_KINDS = {'PM':'POW','TF':'FLT','TL':'TLS'}
+    PM_CHANNELS = [1,2]
+    PM_UNIT = ['W','mW','dBm']
+    STATUS = {'Off':0,'On':1}
+
+
+    def __init__(self, resourceName, **kwargs):
+        super(ApexAP1000, self).__init__(
+            resourceName,
+            "Apex AP1000",
+            **kwargs
+        )
+        self.__slot = None
+
+    def headStr(self,kind):
+        return ApexAP1000.DEV_KINDS[kind]+"[%02d]:"%self.slot
+
+    @property
+    def slot(self):
+        """ Get current slot number
+        """
+        if self.__slot in ApexAP1000.SLOTS:
+            return self.__slot
+        else:
+            raise ValueError('Bad slot number !')
+
+    @slot.setter
+    def slot(self,num):
+        """ Set current slot number
+        You have to set the good slot before asking the functions related
+        to any module
+        """
+        if num in ApexAP1000.SLOTS:
+            self.__slot=num
+        else:
+            raise ValueError('Bad slot number !')
+
+    # Powermeter module related functions
+    @property
+    def PM_averages(self):
+        """ PowerMeter module.
+        Get wanted number of averages for current slot.
+        """
+        return int(self.ask(self.headStr('PM')+'AVG?'))
+
+    @PM_averages.setter
+    def PM_averages(self,num):
+        """ PowerMeter module.
+        Sets wanted number of averages for current slot.
+        """
+        self.write(self.headStr('PM')+'AVG %d'%num)
+
+    def PM_getPower(self,channel,unit='W'):
+        """ PowerMeter module.
+        Get actual power on given channel with given unit.
+        - channel can be {}.
+        - unit can be {}.
+        """
+        if unit not in ApexAP1000.PM_UNIT:
+            raise ValueError('Unknow physical unit during power measurement')
+        if channel not in ApexAP1000.PM_CHANNELS:
+            raise ValueError('Unknow channel during power measurement')
+        str = {'W':'MW','mW':'MW','dBm':'DBM'}
+        value = float(self.ask(self.headStr('PM')+'%s[%d]?'%(str[unit],channel)))
+        if unit is 'W':
+            value = value * 1e-3
+        return value
+
+    def PM_setWavelength(self,channel,wavelength):
+        """ PowerMeter module.
+        Sets wavelength for given channel for calibration
+        - channel can be {}.
+        """
+        if channel not in ApexAP1000.PM_CHANNELS:
+            raise ValueError('Unknow channel during power measurement')
+        sentStr = self.headStr('PM')+'SETWAVELENGTH[%d] %g'%(channel,wavelength)
+        return self.write(sentStr)
+
+    # tunablefilter module related functions
+    @property
+    def TF_wavelength(self):
+        """ Tunable Filter module.
+        Gets tunable filter wavelength.
+        """
+        return int(self.ask(self.headStr('TF')+'TWL?'))
+
+    @TF_wavelength.setter
+    def TF_wavelength(self,wavelength):
+        """ Tunable Filter  module.
+        Sets tunable filter wavelength.
```

Comment:
It would be nice if you can comment on the valid values and units for setting (and getting) the wavelength. Maybe also consider using a validator. (This comment also holds for other functions/properties)